src/main/cpp/src/hive_hash.cu

```diff
@@ -161,8 +162,16 @@ hive_hash_value_t __device__ inline hive_hash_function<cudf::timestamp_us>::oper
 template <template <typename> class hash_function, typename Nullate>
 class hive_device_row_hasher {
  public:
-  CUDF_HOST_DEVICE hive_device_row_hasher(Nullate check_nulls, cudf::table_device_view t) noexcept
-    : _check_nulls{check_nulls}, _table{t}
+  CUDF_HOST_DEVICE hive_device_row_hasher(Nullate check_nulls,
+                                          cudf::table_device_view t,
+                                          cudf::column_device_view* flattened_column_views,
+                                          cudf::size_type* first_child_index,
+                                          cudf::size_type* nested_column_map) noexcept
```

Comment:
Now we no longer need `table_device_view t` for hashing. All the columns are stored in `flattened_column_views`. We only need to know the number of input columns.
In addition, I don't think this functor is constructed in device code.
```suggestion
  hive_device_row_hasher(Nullate check_nulls,
                                          cudf::column_device_view* flattened_column_views,
                                          cudf::size_type* first_child_index,
                                          cudf::size_type* nested_column_map,
                                          cudf::size_type num_columns) noexcept
```

Reply:
The `flattened_column_views` only contains nested columns and columns obtained by flattening the nested columns. I did not change the calculation logic for the basic types. Sorry that my variable names did not reflect this.

Reply:
Okay then it is another problem that needs to be fixed. We should not move back and forth between two places to access nested/non nested columns. That is a bad memory access pattern. We need a coalesce pattern. Let's flatten all columns into one array. So we will have `flattened_cviews` array storing all columns but the nested columns are empty out.